File: gempak/source/gemlib/mv/Linux/mvswp4.f
	FUNCTION MV_SWP4  ( n, valin, valout )
C************************************************************************
C* MV_SWP4								*
C*									*
C* This function converts the byte order of each word from 1-2-3-4 to	*
C* 4-3-2-1.								*
C*									*
C* INTEGER MV_SWP4  ( N, VALIN, VALOUT )				*
C*									*
C* Input parameters:							*
C*	N		INTEGER		Number of values to convert	*
C*	VALIN (N)	BYTE		Input data			*
C*									*
C* Output parameters:							*
C*	VALOUT (N)	BYTE		Converted data			*
C*	MV_SWP4		INTEGER		Return code			*
C*					 0 = normal return		*
C*					>0 = # of invalid inputs	*
C**									*
C* Log:									*
C* K. Brill/NMC		 2/91						*
C************************************************************************
        BYTE            valin (*), valout (*)
C*
        BYTE            temp
C------------------------------------------------------------------------
C*      Loop through the words swapping the byte order from 1-2-3-4 to
C*      4-3-2-1.
C
        index = 1
        DO  i = 1, n
            temp = valin ( index )
            valout ( index ) = valin ( index + 3 )
            valout ( index + 3 ) = temp
            temp = valin ( index + 1 )
            valout ( index + 1 ) = valin ( index + 2 )
            valout ( index + 2 ) = temp
            index = index + 4
        END DO
C*
        MV_SWP4 = 0
C*
        RETURN
        END           

File: gempak/utilities/a2conv/source/mvswp4.c
#include <stdio.h>

int mv_swp4 ( int *n, void *valin, void *valout) 
				 	
/************************************************************************
* MV_SWP4								*
*									*
* This function converts the byte order of each word from 1-2-3-4 to	*
* 4-3-2-1.								*
*									*
* INTEGER MV_SWP4  ( N, VALIN, VALOUT )					*
*									*
* Input parameters:							*
*	N		INTEGER		Number of values to convert	*
*	VALIN (N)	BYTE		Input data			*
*									*
* Output parameters:							*
*	VALOUT (N)	BYTE		Converted data			*
*	MV_SWP4		INTEGER		Return code			*
*					 0 = normal return		*
*					>0 = # of invalid inputs	*
*									*
* Log:									*
* Q. Zhou/Chug	        1/10		Re-write the fortran function   *
*                                       of the same name        	*
************************************************************************/
{
        unsigned char  *vali, *valo, *p;
        unsigned char  temp;
	int i=0;
	

/*      Loop through the words swapping the byte order from 1-2-3-4 to
        4-3-2-1. */
	vali = (unsigned char*) valin;
	valo = (unsigned char* )valout;
	p=	valout;
	while (i < *n) {
		temp = *vali;
		*valo = *(vali+3);
		*(valo +3) = temp;
		temp = *(vali+1);
		*(valo+1) = *(vali+2);
		*(valo+2) = temp;

		vali += 4;
		valo += 4;		
    		i++;
    	}

	valout = (void* )p;
	
        return 0;  
}       

/*int main ( int argc, char **argv )
{
int n=1;
int t=100;
int x,y;
mv_swp4 ( &n, &t, &x);
mv_swp4 ( &n, &x, &y);
    
printf ( "Enter81: %d\n", t );
printf ( "Enter81: %d\n", x );
printf ( "Enter81: %d\n", y );

    return ( 0 );

}*/
